test/tools/ossfuzz/protoToYul.cpp

```diff
@@ -50,6 +50,41 @@ string ProtoConverter::createAlphaNum(string const& _strBytes) const
 	return tmp;
 }
 
+bool ProtoConverter::isCaseLiteralUnique(Literal const& _x)
+{
+	std::string tmp;
+	bool isUnique = false;
+	bool isEmptyString = false;
+	switch (_x.literal_oneof_case())
+	{
+		case Literal::kIntval:
+			tmp = std::to_string(_x.intval());
+			break;
+		case Literal::kHexval:
+			tmp = "0x" + createHex(_x.hexval());
+			break;
+		case Literal::kStrval:
+			tmp = createAlphaNum(_x.strval());
+			if (tmp.empty())
+			{
+				isEmptyString = true;
+				tmp = std::to_string(0);
+			}
+			else
+				tmp = "\"" + tmp + "\"";
+			break;
+		case Literal::LITERAL_ONEOF_NOT_SET:
+			tmp = std::to_string(1);
```

Comment:
Does that mean that the "default" case has value 1? Does that also mean that the following code will never be generated:
```
switch x
case 1 { ... }
default { ... }
```
?

Reply:
Shouldn't you just return `false` here without adding the case value to the set?

Reply:
> Does that mean that the "default" case has value 1?

No, the grammar for default statement is "default BLOCK", so it visits a `Block` instead. This special casing for unset literals is needed because we use the `oneof` directive from the protobuf language which means `at most one of`. So it could happen that the protobuf mutator synthesizes a case statement with an unset string literal.

If we return `false` here, then it could happen that we generate a syntactically incorrect snippet like so
```
// switch statement has been visited because there is exactly one case statement
switch x
// case statement is not visited because it's literal is unset
```

and then we run into an unparesable switch statement. Note that `default` case is optional in our spec and not required. If it were changed to `required` (which means that every switch stmt should have a default case) then we could return false. I think we should keep `default` to optional because it exercises the edge-case when a default statement is (correctly) missing.

Reply:
May be I should clarify. The `oneof { A, B, C}` directive in protobuf leads to the four cases `{A, B, C, None}`. Unset Literals have nothing to do with default case statement i.e., None does not imply default case. Rather, it is referring to a case statement with an unset literal.